### PR TITLE
Bring Device Streaming code into the public repo preview branch

### PR DESCRIPTION
**Pull request by ewertons**
## Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.
Against preview branch.

## Description of the changes
This change brings the code from the Device Streaming private repo ('parent-master' branch) into the public repo 'preview' branch, for public preview of the feature.

## Reference/Link to the issue solved with this PR (if any)
None.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -21,19 +20,22 @@ For IoT Hub Management SDK in .NET, please visit [azure-sdk-for-net](https://git
 Visit [Azure IoT Dev Center][iot-dev-center] to learn more about developing applications for Azure IoT.
 
 ## Samples
-Samples are available at [Azure IoT Samples for C#](https://github.com/Azure-Samples/azure-iot-samples-csharp)
+In the repository, you will find a set of samples that will help you get started:
+* [IoT Hub Device SDK samples](./iothub/device/samples)
```
Comment (prmathur-microsoft) [inline readme.md]: Why is this changed

Comment (prmathur-microsoft) [reply readme.md]: You have to revert this. this is incorrect

Comment (prmathur-microsoft) [reply readme.md]: I think you picked up readme from your private repo and changes in latest is what you need. Can you please revert readme to what is reflected in master?

Comment (ewertons) [reply readme.md]: Fixed.
